x-pack/plugins/synthetics/server/common/mem_writable.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger } from '@kbn/logging';
+import archiver from 'archiver';
+import { Writable, WritableOptions } from 'node:stream';
+
+export class MemWritable extends Writable {
+  private _buffer: Buffer;
+  constructor(opts?: WritableOptions) {
+    super(opts);
+    this._buffer = Buffer.from('');
+  }
+
+  public get buffer(): Buffer {
+    return this._buffer;
+  }
+
+  _write(
```

Comment:
Better to do this way instead of concatenating buffers on every chunk. 
```
  data: Buffer[] = [];
  _write(chunk: Buffer, _encoding: string, callback: () => void) {
    this.data.push(chunk);
    callback();
  }
  toBuffer() {
    return Buffer.concat(this.data).toString('base64')
  }

```

Reply:
👍 like this better

Reply:
Alright, I implemented this pattern but not exactly to spec. `_write` doesn't receive a string, it tends to receive buffers. So I've done something like:

```typescript
  public get buffer(): Buffer {
    return Buffer.concat(this.queue);
  }

  _write(
    chunk: Buffer,
    _encoding: BufferEncoding,
    callback: (error?: Error | null | undefined) => void
  ): void {
    this.queue.push(chunk);
    callback();
  }
```

Still nicer, and gets rid of the `try...catch` from the previous comment as I don't think we need it for a simple push call.

Reply:
[9de233d](https://github.com/elastic/kibana/pull/176094/commits/9de233d712a0cc7532dcb93aa17ca7d5dcab38eb)